src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/Symbols/ICodeGenerationParameterSymbol.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CodeGeneration;
+
+/// <summary>
+/// This interface provides a subset of <see cref="IParameterSymbol"/> for use in code generation scenarios.
+/// </summary>
+internal interface ICodeGenerationParameterSymbol : ICodeGenerationSymbol
```

Comment:
i'm not really understanding why we need the ICodeGen versions.  If the symbol mapping factory can make instances of the actual interface type, then why not have everything operate in only taht space?

Reply:
Basically, why not *soley* have the CodeGenTypes, and then have the helpers that get you the core compiler interface form from those?  i don't see what the ICodeGenXXX types do for us.  Cna you clarify why they are needed?

Reply:
We would lose some validation support if we took this approach (signature mismatch in the implementation would be a silent failure instead of an early failure.